### DISCUSSION TITLE
OpenApi: Make sure that component schemas are never null

### DISCUSSION THREAD

**Discussion by Thijmen** [dotnet/aspnetcore#59397] [Q&A]
Hi there,

I am working around this issue that I am having. For some reason, it marks some component schemas as nullable, which I do not understand. In my opinion (correct my if I am wrong), the schema should never be null, but the properties referencing that schema may be. How can I solve this? 

Example:
```c#
public class A
{
    public required Guid Id { get; init; }
    public B? B { get; init; }
}

public class B
{
    public required Guid Id { get; init; }
    public required string Name { get; init; }
}
```

Having such, leads to following (partial) schema;

```json
"components": {
    "schemas": {
      "A": {
        "required": [
          "id"
        ],
        "type": "object",
        "properties": {
          "id": {
            "type": "string",
            "format": "uuid"
          },
          "b": {
            "$ref": "#/components/schemas/B"
          }
        }
      },
      "B": {
        "required": [
          "id",
          "name"
        ],
        "type": "object",
        "properties": {
          "id": {
            "type": "string",
            "format": "uuid"
          },
          "name": {
            "type": "string"
          }
        },
        "nullable": true
      },
```

Is there a way to have "B" schema to not be nullable?

**Comment by Thijmen** [comment]
@martincostello since you seem to know the internals pretty well - any idea? I do not see why schema B is nullable but schema A is not. I just need to make schema B not nullable :-)

**Reply by martincostello** [reply]
This comment _might_ answer your question: https://github.com/dotnet/aspnetcore/issues/59056#issuecomment-2523233653
